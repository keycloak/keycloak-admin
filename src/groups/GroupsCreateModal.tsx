import React, { useContext } from "react";
import {
  AlertVariant,
  Button,
  Form,
  FormGroup,
  Modal,
  ModalVariant,
  TextInput
} from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import { HttpClientContext } from "../context/http-service/HttpClientContext";
import { RealmContext } from "../context/realm-context/RealmContext";
import { useAlerts } from "../components/alert/Alerts";
import { useForm } from "react-hook-form";

type GroupsCreateModalProps = {
  handleModalToggle: () => void;
  isCreateModalOpen: boolean;
  setIsCreateModalOpen: (isCreateModalOpen: boolean) => void;
  createGroupName: string;
  setCreateGroupName: (createGroupName: string) => void;
  refresh: () => void;
};

export const GroupsCreateModal = ({
  handleModalToggle,
  isCreateModalOpen,
  setIsCreateModalOpen,
  createGroupName,
  setCreateGroupName,
  refresh
}: GroupsCreateModalProps) => {
  const { t } = useTranslation("groups");
  const httpClient = useContext(HttpClientContext)!;
  const { realm } = useContext(RealmContext);
  const { addAlert } = useAlerts();
  const form = useForm();
  const { register, errors } = form;

  // eslint-disable-next-line no-shadow
  const valueChange = (createGroupName: string) => {
    setCreateGroupName(createGroupName);
  };

  const submitForm = async () => {
    if (await form.trigger()) {
      try {
        await httpClient.doPost(`/admin/realms/${realm}/groups`, {
          name: createGroupName
        });
        setIsCreateModalOpen(false);
        setCreateGroupName("");
        refresh();
        addAlert(t("groupCreated"), AlertVariant.success);
      } catch (error) {
        addAlert(
          `${t("couldNotCreateGroup")} ': '${error}'`,
          AlertVariant.danger
        );
      }
    }
  };

  return (
    <>
      <Modal
        variant={ModalVariant.small}
        title={t("createAGroup")}
        isOpen={isCreateModalOpen}
        onClose={handleModalToggle}
        actions={[
          <Button key="confirm" variant="primary" onClick={() => submitForm()}>
            {t("create")}
          </Button>
        ]}
      >
        <Form isHorizontal>
          <FormGroup
            name="create-modal-group"
            label={t("name")}
            fieldId="group-id"
            helperTextInvalid={t("common:required")}
            validated={errors.name ? "error" : "default"}
            isRequired
          >
            <TextInput
              ref={register({ required: true })}
              type="text"
              id="create-group-name"
              name="name"
              value={createGroupName}
              onChange={valueChange}
            />
          </FormGroup>
        </Form>
      </Modal>
    </>
  );
};
