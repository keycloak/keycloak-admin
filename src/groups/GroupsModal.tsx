import React from "react";
import {
  AlertVariant,
  Button,
  ButtonVariant,
  Form,
  FormGroup,
  Modal,
  ModalVariant,
  TextInput,
  ValidatedOptions,
} from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import { useForm } from "react-hook-form";

import GroupRepresentation from "keycloak-admin/lib/defs/groupRepresentation";
import { useAdminClient } from "../context/auth/AdminClient";
import { useAlerts } from "../components/alert/Alerts";

type GroupsModalProps = {
  id?: string;
  rename?: string;
  handleModalToggle: () => void;
  refresh: (group?: GroupRepresentation) => void;
};

export const GroupsModal = ({
  id,
  rename,
  handleModalToggle,
  refresh,
}: GroupsModalProps) => {
  const { t } = useTranslation("groups");
  const adminClient = useAdminClient();
  const { addAlert } = useAlerts();
  const { register, errors, handleSubmit } = useForm({
    defaultValues: { name: rename },
  });

  const submitForm = async (group: GroupRepresentation) => {
    try {
      if (!id) {
        await adminClient.groups.create(group);
      } else if (rename) {
        await adminClient.groups.update({ id }, group);
      } else {
        await adminClient.groups.setOrCreateChild({ id }, group);
      }

      refresh(rename ? group : undefined);
      handleModalToggle();
      addAlert(
        t(rename ? "groupUpdated" : "groupCreated"),
        AlertVariant.success
      );
    } catch (error) {
      addAlert(t("couldNotCreateGroup", { error }), AlertVariant.danger);
    }
  };

  return (
    <Modal
      variant={ModalVariant.small}
      title={t(rename ? "renameAGroup" : "createAGroup")}
      isOpen={true}
      onClose={handleModalToggle}
      actions={[
        <Button
          data-testid={`${rename ? "rename" : "create"}Group`}
          key="confirm"
          variant="primary"
          type="submit"
          form="group-form"
        >
          {t(rename ? "rename" : "create")}
        </Button>,
        <Button
          id="modal-cancel"
          key="cancel"
          variant={ButtonVariant.secondary}
          onClick={() => {
            handleModalToggle();
          }}
        >
          {t("common:cancel")}
        </Button>,
      ]}
    >
      <Form id="group-form" isHorizontal onSubmit={handleSubmit(submitForm)}>
        <FormGroup
          name="create-modal-group"
          label={t("common:name")}
          fieldId="group-id"
          helperTextInvalid={t("common:required")}
          validated={
            errors.name ? ValidatedOptions.error : ValidatedOptions.default
          }
          isRequired
        >
          <TextInput
            data-testid="groupNameInput"
            ref={register({ required: true })}
            autoFocus
            type="text"
            id="create-group-name"
            name="name"
            validated={
              errors.name ? ValidatedOptions.error : ValidatedOptions.default
            }
          />
        </FormGroup>
      </Form>
    </Modal>
  );
};
