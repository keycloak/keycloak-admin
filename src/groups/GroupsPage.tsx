import React from "react";

export const GroupsPage = () => {
  return <>The Groups Page</>;
};
