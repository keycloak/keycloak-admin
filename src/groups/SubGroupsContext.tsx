import React, { createContext, ReactNode, useContext, useState } from "react";
import GroupRepresentation from "keycloak-admin/lib/defs/groupRepresentation";

type SubGroupsProps = {
  subGroups: GroupRepresentation[];
  setSubGroups: (group: GroupRepresentation[]) => void;
  clear: () => void;
  remove: (group: GroupRepresentation) => void;
  currentGroup: () => GroupRepresentation;
};

const SubGroupContext = createContext<SubGroupsProps>({
  subGroups: [],
  setSubGroups: () => {},
  clear: () => {},
  remove: () => {},
  currentGroup: () => {
    return {};
  },
});

export const SubGroups = ({ children }: { children: ReactNode }) => {
  const [subGroups, setSubGroups] = useState<GroupRepresentation[]>([]);

  const clear = () => setSubGroups([]);
  const remove = (group: GroupRepresentation) =>
    setSubGroups(
      subGroups.slice(
        0,
        subGroups.findIndex((g) => g.id === group.id)
      )
    );
  const currentGroup = () => subGroups[subGroups.length - 1];
  return (
    <SubGroupContext.Provider
      value={{ subGroups, setSubGroups, clear, remove, currentGroup }}
    >
      {children}
    </SubGroupContext.Provider>
  );
};

export const useSubGroups = () => useContext(SubGroupContext);
