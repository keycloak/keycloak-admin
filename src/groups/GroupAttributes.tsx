import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { useFieldArray, useForm } from "react-hook-form";
import { AlertVariant } from "@patternfly/react-core";

import { useAlerts } from "../components/alert/Alerts";
import {
  arrayToAttributes,
  AttributeForm,
  AttributesForm,
  attributesToArray,
} from "../components/attribute-form/AttributeForm";
import { useAdminClient } from "../context/auth/AdminClient";

import { getLastId } from "./groupIdUtils";
import { useSubGroups } from "./SubGroupsContext";

export const GroupAttributes = () => {
  const { t } = useTranslation("groups");
  const adminClient = useAdminClient();
  const { addAlert } = useAlerts();
  const form = useForm<AttributeForm>({ mode: "onChange" });
  const { fields, append, remove } = useFieldArray({
    control: form.control,
    name: "attributes",
  });

  const id = getLastId(location.pathname);
  const { subGroups } = useSubGroups();
  const getGroup = () => subGroups[subGroups.length - 1];

  useEffect(() => {
    const group = getGroup();
    const attributes = attributesToArray(group.attributes!);
    attributes.push({ key: "", value: "" });
    form.setValue("attributes", attributes);
  }, [id]);

  const save = async (attributeForm: AttributeForm) => {
    try {
      const group = getGroup();
      await adminClient.groups.update(
        { id: id! },
        { ...group, attributes: arrayToAttributes(attributeForm.attributes) }
      );
      addAlert(t("groupUpdated"), AlertVariant.success);
    } catch (error) {
      addAlert(t("groupUpdateError", { error }), AlertVariant.danger);
    }
  };

  return (
    <AttributesForm
      form={form}
      save={save}
      array={{ fields, append, remove }}
      reset={() =>
        form.reset({
          attributes: attributesToArray(
            subGroups[subGroups.length - 1].attributes
          ),
        })
      }
    />
  );
};
