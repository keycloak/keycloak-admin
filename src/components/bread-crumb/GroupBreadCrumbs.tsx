import React, { useEffect } from "react";
import { Link, useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";
import { Breadcrumb, BreadcrumbItem } from "@patternfly/react-core";

import { useSubGroups } from "../../groups/SubGroupsContext";
import { useRealm } from "../../context/realm-context/RealmContext";

export const GroupBreadCrumbs = () => {
  const { t } = useTranslation();
  const { clear, remove, subGroups } = useSubGroups();
  const { realm } = useRealm();

  const history = useHistory();

  useEffect(() => {
    return history.listen(({ pathname }) => {
      if (pathname.indexOf("/groups") === -1 || pathname.endsWith("/groups")) {
        clear();
      }
    });
  }, [history]);

  return (
    <>
      {subGroups.length !== 0 && (
        <Breadcrumb>
          <BreadcrumbItem key="home">
            <Link to={`/${realm}/groups`}>{t("groups")}</Link>
          </BreadcrumbItem>
          {subGroups.map((group, i) => (
            <BreadcrumbItem key={i} isActive={subGroups.length - 1 === i}>
              {subGroups.length - 1 !== i && (
                <Link
                  to={location.pathname.substr(
                    0,
                    location.pathname.indexOf(group.id!) + group.id!.length
                  )}
                  onClick={() => remove(group)}
                >
                  {group.name}
                </Link>
              )}
              {subGroups.length - 1 === i && <>{group.name}</>}
            </BreadcrumbItem>
          ))}
        </Breadcrumb>
      )}
    </>
  );
};
