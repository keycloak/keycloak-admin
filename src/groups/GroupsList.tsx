import React, { useState, useEffect, useContext } from "react";
import {
  Table,
  TableHeader,
  TableBody,
  TableVariant,
} from "@patternfly/react-table";
import { Button } from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import { GroupRepresentation } from "./models/groups";
import { UsersIcon } from "@patternfly/react-icons";
import { HttpClientContext } from "../context/http-service/HttpClientContext";

type GroupsListProps = {
  list?: GroupRepresentation[];
};

export const GroupsList = ({ list }: GroupsListProps) => {
  const { t } = useTranslation("groups");
  const httpClient = useContext(HttpClientContext)!;
  const columnGroupName: keyof GroupRepresentation = "name";
  const columnGroupNumber: keyof GroupRepresentation = "membersLength";
  const [formattedData, setFormattedData] = useState([
    { cells: [<Button key="0">Test</Button>], selected: false },
  ]);

  const formatData = (data: GroupRepresentation[]) =>
    data.map((group: { [key: string]: any }, index) => {
      const groupName = group[columnGroupName];
      const groupNumber = group[columnGroupNumber];
      return {
        cells: [
          <Button variant="link" key={index}>
            {groupName}
          </Button>,
          <div className="keycloak-admin--groups__member-count" key={index}>
            <UsersIcon />
            {groupNumber}
          </div>,
        ],
        selected: false,
      };
    });

  useEffect(() => {
    setFormattedData(formatData(list!));
  }, [list]);

  function onSelect(
    event: React.FormEvent<HTMLInputElement>,
    isSelected: boolean,
    rowId: number
  ) {
    let localRow;
    if (rowId === undefined) {
      localRow = formattedData.map((row: { [key: string]: any }) => {
        row.selected = isSelected;
        return row;
      });
    } else {
      localRow = [...formattedData];
      localRow[rowId].selected = isSelected;
      setFormattedData(localRow);
    }
  }

  // Delete individual rows using the action in the table
  function onDelete(rowIndex: number) {
    const localFilteredData = [...list!];
    httpClient.doDelete(
      `/admin/realms/master/groups/${localFilteredData[rowIndex].id}`
    );
    // TO DO update the state
  }

  const tableHeader = [{ title: t("groupName") }, { title: t("members") }];
  const actions = [
    {
      title: t("moveTo"),
      onClick: () => console.log("TO DO: Add move to functionality"),
    },
    {
      title: t("common:Delete"),
      onClick: () => onDelete,
    },
  ];

  return (
    <React.Fragment>
      {formattedData && (
        <Table
          actions={actions}
          variant={TableVariant.compact}
          onSelect={onSelect}
          canSelectAll={false}
          aria-label={t("tableOfGroups")}
          cells={tableHeader}
          rows={formattedData}
        >
          <TableHeader />
          <TableBody />
        </Table>
      )}
    </React.Fragment>
  );
};
