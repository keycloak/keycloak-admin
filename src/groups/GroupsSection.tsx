import React, { useEffect, useState } from "react";
import { useHistory, useLocation } from "react-router-dom";
import { useTranslation } from "react-i18next";
import { useErrorHandler } from "react-error-boundary";
import {
  DropdownItem,
  PageSection,
  PageSectionVariants,
  AlertVariant,
  Tab,
  TabTitleText,
  Tabs,
} from "@patternfly/react-core";
import GroupRepresentation from "keycloak-admin/lib/defs/groupRepresentation";

import { ViewHeader } from "../components/view-header/ViewHeader";
import { asyncStateFetch, useAdminClient } from "../context/auth/AdminClient";
import { useAlerts } from "../components/alert/Alerts";
import { useRealm } from "../context/realm-context/RealmContext";

import { useSubGroups } from "./SubGroupsContext";
import { GroupTable } from "./GroupTable";
import { getId, getLastId } from "./groupIdUtils";
import { Members } from "./Members";
import { GroupAttributes } from "./GroupAttributes";
import { GroupsModal } from "./GroupsModal";

import "./GroupsSection.css";

export const GroupsSection = () => {
  const { t } = useTranslation("groups");
  const [activeTab, setActiveTab] = useState(0);

  const adminClient = useAdminClient();
  const { subGroups, setSubGroups, currentGroup } = useSubGroups();
  const { addAlert } = useAlerts();
  const { realm } = useRealm();
  const errorHandler = useErrorHandler();

  const [rename, setRename] = useState<string>();

  const history = useHistory();
  const location = useLocation();
  const id = getLastId(location.pathname);

  const deleteGroup = async (group: GroupRepresentation) => {
    try {
      await adminClient.groups.del({
        id: group.id!,
      });
      addAlert(t("groupDelete"), AlertVariant.success);
    } catch (error) {
      addAlert(t("groupDeleteError", { error }), AlertVariant.danger);
    }
    return true;
  };

  useEffect(
    () =>
      asyncStateFetch(
        async () => {
          const ids = getId(location.pathname);
          const isNavigationStateInValid =
            ids && ids.length !== subGroups.length + 1;
          if (isNavigationStateInValid) {
            const groups: GroupRepresentation[] = [];
            for (const i of ids!) {
              const group = await adminClient.groups.findOne({ id: i });
              if (group) groups.push(group);
            }
            return groups;
          } else {
            if (id) {
              const group = await adminClient.groups.findOne({ id: id });
              if (group) {
                return [...subGroups, group];
              } else {
                return subGroups;
              }
            } else {
              return subGroups;
            }
          }
        },
        (groups: GroupRepresentation[]) => setSubGroups(groups),
        errorHandler
      ),
    [id]
  );

  const SearchDropdown = (
    <DropdownItem
      data-testid="searchGroup"
      key="searchGroup"
      onClick={() => history.push(`/${realm}/groups/search`)}
    >
      {t("searchGroup")}
    </DropdownItem>
  );

  return (
    <>
      {rename && (
        <GroupsModal
          id={id}
          rename={rename}
          refresh={(group) =>
            setSubGroups([...subGroups.slice(0, subGroups.length - 1), group!])
          }
          handleModalToggle={() => setRename(undefined)}
        />
      )}
      <ViewHeader
        titleKey="groups:groups"
        subKey={!id ? "groups:groupsDescription" : ""}
        divider={!id}
        dropdownItems={
          id
            ? [
                SearchDropdown,
                <DropdownItem
                  data-testid="renameGroupAction"
                  key="renameGroup"
                  onClick={() => setRename(currentGroup().name)}
                >
                  {t("renameGroup")}
                </DropdownItem>,
                <DropdownItem
                  data-testid="deleteGroup"
                  key="deleteGroup"
                  onClick={() => {
                    deleteGroup({ id });
                    history.push(
                      location.pathname.substr(
                        0,
                        location.pathname.lastIndexOf("/")
                      )
                    );
                  }}
                >
                  {t("deleteGroup")}
                </DropdownItem>,
              ]
            : [SearchDropdown]
        }
      />
      <PageSection variant={PageSectionVariants.light} className="pf-u-p-0">
        {subGroups.length > 0 && (
          <Tabs
            inset={{
              default: "insetNone",
              md: "insetSm",
              xl: "inset2xl",
              "2xl": "insetLg",
            }}
            activeKey={activeTab}
            onSelect={(_, key) => setActiveTab(key as number)}
            isBox
          >
            <Tab
              data-testid="groups"
              eventKey={0}
              title={<TabTitleText>{t("childGroups")}</TabTitleText>}
            >
              <GroupTable />
            </Tab>
            <Tab
              data-testid="members"
              eventKey={1}
              title={<TabTitleText>{t("members")}</TabTitleText>}
            >
              <Members />
            </Tab>
            <Tab
              data-testid="attributes"
              eventKey={2}
              title={<TabTitleText>{t("attributes")}</TabTitleText>}
            >
              <GroupAttributes />
            </Tab>
          </Tabs>
        )}
        {subGroups.length === 0 && <GroupTable />}
      </PageSection>
    </>
  );
};
