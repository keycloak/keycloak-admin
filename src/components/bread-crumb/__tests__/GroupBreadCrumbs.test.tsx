import React, { useEffect } from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import { GroupBreadCrumbs } from "../GroupBreadCrumbs";
import { SubGroups, useSubGroups } from "../../../groups/SubGroupsContext";

const GroupCrumbs = () => {
  const { setSubGroups } = useSubGroups();
  useEffect(() => {
    setSubGroups([
      { id: "1", name: "first group" },
      { id: "123", name: "active group" },
    ]);
  }, []);

  return <GroupBreadCrumbs />;
};

describe("Group BreadCrumbs tests", () => {
  it("couple of crumbs", () => {
    const crumbs = mount(
      <MemoryRouter initialEntries={["/groups"]}>
        <SubGroups>
          <GroupCrumbs />
        </SubGroups>
      </MemoryRouter>
    );

    expect(crumbs.find(GroupCrumbs)).toMatchSnapshot();
  });
});
