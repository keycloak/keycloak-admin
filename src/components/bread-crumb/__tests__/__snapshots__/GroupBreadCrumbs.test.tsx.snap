// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group BreadCrumbs tests couple of crumbs 1`] = `
<GroupCrumbs>
  <GroupBreadCrumbs>
    <Breadcrumb>
      <nav
        aria-label="Breadcrumb"
        className="pf-c-breadcrumb"
        data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
        data-ouia-component-type="PF4/Breadcrumb"
        data-ouia-safe={true}
      >
        <ol
          className="pf-c-breadcrumb__list"
        >
          <BreadcrumbItem
            key=".$home"
            showDivider={false}
          >
            <li
              className="pf-c-breadcrumb__item"
            >
              <Link
                to="//groups"
              >
                <LinkAnchor
                  href="//groups"
                  navigate={[Function]}
                >
                  <a
                    href="//groups"
                    onClick={[Function]}
                  >
                    Groups
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </BreadcrumbItem>
          <BreadcrumbItem
            isActive={false}
            key=".1:$0"
            showDivider={true}
          >
            <li
              className="pf-c-breadcrumb__item"
            >
              <span
                className="pf-c-breadcrumb__item-divider"
              >
                <AngleRightIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </AngleRightIcon>
              </span>
              <Link
                onClick={[Function]}
                to=""
              >
                <LinkAnchor
                  href="/"
                  navigate={[Function]}
                  onClick={[Function]}
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    first group
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </BreadcrumbItem>
          <BreadcrumbItem
            isActive={true}
            key=".1:$1"
            showDivider={true}
          >
            <li
              className="pf-c-breadcrumb__item"
            >
              <span
                className="pf-c-breadcrumb__item-divider"
              >
                <AngleRightIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </AngleRightIcon>
              </span>
              Group details
            </li>
          </BreadcrumbItem>
        </ol>
      </nav>
    </Breadcrumb>
  </GroupBreadCrumbs>
</GroupCrumbs>
`;
